resources/devbox/macos/macos.sh

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env bash
+
+set -e
+set -u
+
+## Note: does not use standard build script include
+# adjust relative paths as necessary. Note that this will not work in a symlinked path reliably
+THIS_SCRIPT="${BASH_SOURCE[0]}"
+THIS_DIR=$(dirname "$THIS_SCRIPT")
+
+# Determine which components to install
+REQUIRE_ANDROID=false
+REQUIRE_IOS=false
+REQUIRE_MACOS=false
+REQUIRE_WEB=false
+
+# Optional components
+REQUIRE_KMCOMP=false
+REQUIRE_PANDOC=false
+REQUIRE_SENTRYCLI=false
+
+# Parse args
+shopt -s nocasematch
+
+while [[ $# -gt 0 ]] ; do
+    key="$1"
+    case $key in
+        --help)
+            echo "Usage: macos.sh targets"
+            echo "  targets: one or more of: "
+            echo "             android ios macos web all"
+            echo "  optional targets:"
+            echo "             kmcomp pandoc sentry-cli"
```

Comment:
I assume these "optional" targets are included in "all"?  Could be a bit more explicit about that if so, and a _lot_ more explicit if not.

Based on what follows, looks like it's the latter.  Could we also have an `all-optional` shorthand?